server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -477,6 +479,39 @@ public void testSegmentsWithIndexSort() throws Exception {
         }
     }
 
+    public void testSegmentsWithNestedFieldIndexSort() throws Exception {
+        Sort indexSort = new Sort(new SortedSetSortField("contacts.name", false));
+        try (
+            Store store = createStore();
+            Engine engine = createEngine(defaultSettings, store, createTempDir(), NoMergePolicy.INSTANCE, null, null, null, indexSort, null)
+        ) {
+            List<Segment> segments = engine.segments(true);
+            assertThat(segments.isEmpty(), equalTo(true));
+
+            ParsedDocument doc1 = createDocumentWithNestedField("1", "Alice", 30);
```

Comment:
This doc may not test `nested` type where we have multiple Lucene docs. Can we modify the same to send list of docs to Lucene?